python/cugraph/cugraph/gnn/pyg_extensions/data/cugraph_store.py

```diff
@@ -433,159 +469,222 @@ def neighbor_sample(
             [sampling_results.destinations, sampling_results.sources]
         ).unique()
 
-        noi = self.__graph.get_vertex_data(
-            nodes_of_interest.compute() if self.is_mg else nodes_of_interest,
-            columns=[self.__graph.vertex_col_name, self.__graph.type_col_name]
+        if self.is_mg:
+            nodes_of_interest = nodes_of_interest.compute()
+
+        nodes_of_interest = nodes_of_interest.sort_values()
+
+        # Get the node index (for creating the edge index),
+        # the node type groupings, and the node properties.
+        noi_index, noi_groups, noi_tensors = (
+            self.__get_renumbered_vertex_data_from_sample(
+                nodes_of_interest.values_host if self.is_mg
+                else nodes_of_interest
+            )
         )
+        del nodes_of_interest
 
-        noi_types = noi[self.__graph.type_col_name].unique()
-        noi = noi.groupby(self.__graph.type_col_name)
+        # Get the new edge index (by type as expected for HeteroData)
+        # FIXME handle edge ids
+        row_dict, col_dict = self.__get_renumbered_edges_from_sample(
+            sampling_results,
+            noi_index
+        )
 
-        if self.is_mg:
-            noi_types = noi_types.compute()
-        noi_types = noi_types.to_pandas()
+        return (noi_groups, row_dict, col_dict, noi_tensors)
 
-        # these should contain the original ids, they will be auto-renumbered
+    def __get_renumbered_vertex_data_from_sample(self, nodes_of_interest):
+        """
+        nodes_of_interest must be sorted
```

Comment:
Does the input have to be sorted or you want results to correspond to `nodes_of_interest` ?

Reply:
The input has to be sorted, otherwise the output will be garbage.  Line 475 sorts the nodes of interest prior to passing them to this funciton.

Reply:
But I could do the sorting in this function.  I will go ahead and make the change.

Reply:
done